### PR TITLE
Terms of service

### PR DESCRIPTION
**Pull request by Allfd**
Adds terms of service / privacy policy feature.

Mostly administrative.

### CODE REVIEW

**Comment by DarkPyrolord** [comment]
@tigercat2000 @Fox-McCloud @Crazylemon64 
Has the all clear from me, merge if things are good code-wise.

File: SQL/paradise_schema.sql [hunk 1]
```diff
@@ -359,12 +359,12 @@ DROP TABLE IF EXISTS `privacy`;
...
 /*!40101 SET character_set_client = utf8 */;
 CREATE TABLE `privacy` (
-  `id` int(11) NOT NULL AUTO_INCREMENT,
-  `datetime` datetime NOT NULL,
   `ckey` varchar(32) NOT NULL,
-  `option` varchar(128) NOT NULL,
-  PRIMARY KEY (`id`)
-) ENGINE=InnoDB AUTO_INCREMENT=10000 DEFAULT CHARSET=latin1;
+  `datetime` datetime NOT NULL,
+  `consent` varchar(128) NOT NULL,
```
Comment (Birdtalon) [inline SQL/paradise_schema.sql]: If you're just storing a string of `SIGNED` or `NOTSIGNED` you probably don't need a field this long.

File: code/modules/mob/new_player/new_player.dm [hunk 2]
```diff
@@ -19,7 +20,38 @@
 
 /mob/new_player/verb/new_player_panel()
 	set src = usr
-	new_player_panel_proc()
+	if(tos_consent)
```
Comment (tigercat2000) [inline code/modules/mob/new_player/new_player.dm]: The way this works might cause issues. I'd recommend doing it like this:
```dm
/mob/new_player/verb/new_player_panel()
	set src = usr

	if(handle_tos_consent())
		new_player_panel_proc()

/mob/new_player/proc/handle_tos_consent()
	if(!GLOB.join_tos)
		return TRUE

	establish_db_connection()
	if(!dbcon.IsConnected())
		tos_consent = 1
		return TRUE

	var/DBQuery/query = dbcon.NewQuery("SELECT * FROM [format_table_name("privacy")] WHERE ckey='[src.ckey]' AND consent=1")
	query.Execute()
	while(query.NextRow())
		tos_consent = TRUE
		return TRUE

	privacy_consent()
	return FALSE
```

**Comment by Deelite34** [comment]
Ready up panel in lobby screen does not pop up, in 2nd game lobby after accepting terms, making game unplayable, needs fixing.

**Comment by tigercat2000** [comment]
@Deelite34 Should be fixed.

**Pull request merged**